modules/codegen/src/main/scala/com/twilio/guardrail/core/CoreTermInterp.scala

```diff
@@ -143,25 +142,24 @@ class CoreTermInterp[L <: LA](
         Paths.get(specPath), {
           swagger =>
             try {
-              val Sw = SwaggerTerms.swaggerTerm[L, CodegenApplication[L, ?]]
-              val Sc = LanguageTerms.languageTerm[L, CodegenApplication[L, ?]]
+              import targetInterpreter._
+              val Sw     = implicitly[SwaggerTerms[L, Target]]
+              val Sc     = implicitly[LanguageTerms[L, Target]]
+              val MonadF = () // TODO: Remove me, resolving implicit collision for Monad[Target]
               val program = for {
                 _                  <- Sw.log.debug("Running guardrail codegen")
                 definitionsPkgName <- Sc.fullyQualifyPackageName(pkgName)
                 (proto, codegen) <- Common
-                  .prepareDefinitions[L, Free[CodegenApplication[L, ?], ?]](
+                  .prepareDefinitions[L, Target](
                     kind,
                     context,
                     Tracker(swagger),
                     definitionsPkgName ++ ("definitions" :: dtoPackage)
                   )
                 result <- Common
-                  .writePackage[L, Free[CodegenApplication[L, ?], ?]](proto, codegen, context)(Paths.get(outputPath), pkgName, dtoPackage, customImports)
+                  .writePackage[L, Target](proto, codegen, context)(Paths.get(outputPath), pkgName, dtoPackage, customImports)
               } yield result
-              // Issue #496: Injected StructuredLogger too slow
-              // GuardrailFreeHacks.injectLogs(program, Set("LogPush", "LogPop", "LogDebug", "LogInfo", "LogWarning", "LogError"), Sw.log.push, Sw.log.pop, Free.pure(()))
               program
```

Comment:
Remove the temporary and just return the result of the `for` comprehension directly?